packages/core/src/StateMachine.ts

```diff
@@ -525,6 +529,9 @@ export class StateMachine<
     return restoredSnapshot;
   }
 
-  /** @deprecated an internal property that was acting as a "phantom" type, it's not used by anything right now but it's kept around for compatibility reasons */
+  /**
+   * @internal
```

Comment:
> but it's kept around for compatibility reasons

this is the important bit... ;p by marking it as `@internal` you essentially remove this from the generated declaration files and for now it was kept here to avoid breaking people when they install a new version of XState v5 without installing a new version of `@xstate/react@^4`